### ISSUE TITLE
Adding aria-label attribute for checkbox component (for screen readers)

### ISSUE THREAD

**Issue by mattcfilbert** [uber/baseweb#3905]
The checkbox component does not have an aria-label attribute. Adding one would make it more accessible for screen readers as the title attribute is often insufficient.
## Feature description
Adding a single attribute/prop (aria-label) to the checkbox component to increase accessbility.
